webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/reports/ReportingServiceImpl.scala

```diff
@@ -542,15 +549,19 @@ trait CachedFindRuleNodeStatusReports
    * This is handled in higher level of the app and leads to "no data available" in
    * place of compliance bar.
    */
-  private[this] def checkAndGetCache(nodeIdsToCheck: Set[NodeId]): IOResult[Map[NodeId, NodeStatusReport]] = {
+  private[this] def checkAndGetCache(
+      nodeIdsToCheck: Set[NodeId]
+  )(implicit qc:      QueryContext): IOResult[Map[NodeId, NodeStatusReport]] = {
     if (nodeIdsToCheck.isEmpty) {
       Map[NodeId, NodeStatusReport]().succeed
     } else {
       val now = DateTime.now
 
       for {
         // disabled nodes are ignored
-        allNodeIds   <- nodeInfoService.getAll().map(_.filter { case (_, n) => n.state != NodeState.Ignored }.keySet)
+        allNodeIds   <- nodeFactRepository
+                          .getAll()
+                          .map(_.collect { case (_, n) if (n.rudderSettings.state != NodeState.Ignored) => n.id }.toSet)
```

Comment:
(refactoring): we don't need to create a full collection with filter since we just want the keys, collect is better